### ISSUE TITLE
Limit what options may go into an ackrc file

### ISSUE THREAD

**Issue by petdance** [beyondgrep/ack3#157]
The way that ack treats the ackrc file like just another source of command line arguments is dangerous.  There's no legit reason for someone to put, say, `-f` or `-g` or `--match` or `--output` or `-c` or `-x` in their ackrc file.

The list of options that seem sensible to me are: `-i`, `-S`, `--column`, `-s`, `--pager`, `--heading`, `--break`, `--group`, `--proximate`, `--underline`, `--color*`, `--flush`, `--sort-files`, `--ignore*`, `-r`, `-n`, `--follow`, `--type-add`, `--type-set`, `--type`, `-k`, `-print0`, `-A`, `-B`, `-C`.

Basically anything that regulates behavior vs. regulating output should be forbidden.

**Comment by petdance** [comment]
This is related to #57.  The way we decide what options are mutex is tied to groups of options and their meaning.

**Comment by petdance** [comment]
Certain options are "actions", for lack of a better word.  These can only be on the command line.

`-v`
`-Q`
`--lines`
`--match`
`-l`
`-L`
`-o`
`--output`
`--passthru`
`-c`
`-f`
`-g`
`-print0` (only works with other actions)
`--show-types`
`--files-from`
`-x`
`--help-types`
`--version`
`--ackrc`
`--create-ackrc`
`--dump`
`--help`
`--man`
`--faq`
`--cookbook`
`--thpppt`
`--bar`
`--cathy`

**Comment by n1vux** [comment]
Makes sense. And not really much extra work since we're already filtering a few for security.

**Comment by n1vux** [comment]
(Although putting `--cathy` or `--thpppt` in a project .ackrc would be an amusing prank.)
